pkcs11/csharp/Sample_Helpers.cs

```diff
@@ -372,14 +379,29 @@ public static void PrintAttributes(List<IObjectAttribute> objectAttributes)
                                 Console.WriteLine(name + " : " + attr.GetValueAsBool().ToString());
                             break;
 
+                        case (uint)CKA.CKA_THALES_CUSTOM_1:
+                        case (uint)CKA.CKA_THALES_CUSTOM_2:
+                        case (uint)CKA.CKA_THALES_CUSTOM_3:
+                        case (uint)CKA.CKA_THALES_CUSTOM_4:
+                        case (uint)CKA.CKA_THALES_CUSTOM_5:
+                            Console.WriteLine(name + " : " + attr.GetValueAsString());
+                            break;
+
                         default:
-                            valArray = attr.GetValueAsByteArray();
-                            str = BitConverter.ToString(valArray);
-                            StringBuilder sb = new StringBuilder();
-                            foreach (var c in str)
-                                if (c != '-') sb.Append(c);
+			    if (attr.GetValueAsByteArray().Length == 0)
```

Comment:
use GetValueAsByteArray() once and use in if and else part.

Reply:
done